core/cont/src/TClassTable.cxx

```diff
@@ -590,15 +662,22 @@ DictFuncPtr_t TClassTable::GetDict(const char *cname)
 
 DictFuncPtr_t TClassTable::GetDict(const std::type_info& info)
 {
-   if (!CheckClassTableInit()) return nullptr;
+   if (!CheckClassTableInit())
+      return nullptr;
+
+   if (gDebug > 9)
+      ROOT::GetROOT(); // Info might recursively call TClassTable during the gROOT init
```

Comment:
> why not moving the ::Info() call here, before taking the lock
> The only issue I see is out-of-order messages in the log,

If 2 threads are calling `GetDict` then there will be zero correlation between the message and the printing of `fgIdMap` making it harder to reason about the failure that might be under investigation.  In my opinion keeping the 2 prints together is worth the slight (because there would still be 2 `if (gDebug > 9)`) increase in readability.